src/libraries/System.Text.Json/ref/System.Text.Json.csproj

```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);netcoreapp3.0;netstandard2.0;net461</TargetFrameworks>
     <Nullable>enable</Nullable>
+    <IncludeInternalObsoleteAttribute>true</IncludeInternalObsoleteAttribute>
```

Comment:
Should a condition be applied here, or instead place one https://github.com/dotnet/runtime/blob/466deef23436fd8705db5ebaa6d719c3fa153766/src/libraries/Directory.Build.targets#L247
so that the attribute isn't included when the framework carries a sufficient copy?

Reply:
As best as I can tell, the condition there in `runtime/src/libraries/Directory.Build.targets` should already be accomplishing that by only applying for `netstandard`, `netcoreapp`, or `.NETFramework`. I think that would exclude `net5.0` and newer.

Is there a condition that I'm missing where you think it could be including the attribute when that's unnecessary? And/or do you think the property should be named such that the condition is connoted?

Reply:
I suppose you're right.  I was thinking of net5.0 as netcoreapp since it's still treated that way behind the scenes, but the way the condition is written it will not apply when folks use `net5.0` and we don't use `netcoreapp5.0` or `netcoreapp6.0` in our repo.

Reply:
FWIW I had to go double-check that we used `net5.0` and `net6.0` (not `netcoreapp5.0` / `netcoreapp6.0`) :wink:. We could potentially simplify this down to checking to see if the target framework version is less than 5.0.

Reply:
That approach was used just a few lines above actually. I do think it's more concise.

```xml
<When Condition="'$(IncludeInternalObsoleteAttribute)' == 'true' and ('$(TargetFrameworkIdentifier)' != '.NETCoreApp' or $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))">
```

This makes me somewhat yearn for the implicitly defined constants of `NET5_0_OR_GREATER` to also be available as properties. 🤔  But I'll queue up a follow-up PR to make this adjustment. I'll wait a bit though to see if you or @layomia have any other feedback.

Reply:
> This makes me somewhat yearn for the implicitly defined constants of NET5_0_OR_GREATER to also be available as properties

Funny: the reason for those constants is that we don't have a complex preprocessor in C#.  MSBuild gives you much more flexibility, but there's less prescriptive patterns to follow.

I'm fine so long as we don't unnecessarily duplicate the attribute definition, which was the case already.  I like the simplified condition in your diff.

Reply:
Thanks. I'll leave this as I have it in that pending PR. We'll keep on eye on this to determine if there are enough scenarios to justify MSBuild properties such as `NET5_0_OR_GREATER` and `NET6_0_OR_GREATER`.